### DISCUSSION TITLE
Due to my network limitations, I don't have a dhcp server, how can I configure static ip into kernel parameters

### DISCUSSION THREAD

**Discussion by bestsiyue** [antonym/netboot.xyz#1209] [Q&A]
What I can know is that centos probably uses this format to write static IP addresses.
  biosdevname=0 net.ifnames=0 ifname=com0:${netX/mac} ip=${ip}::${gateway}:${netmask}::com0:none ipv6.disable=1

But I still can't find a way for debian and ubuntu, and my network address needs a vlan tag, which may be a little more complicated.

In addition, there is a parameter in debian.ipxe that I do not understand, and I have not found its source, can you explain it to me?
${netcfg}

**Comment by n3gwg** [comment]
I am not sure I fully understand your problem though I do think I can help.

Let me first ask, what network limitations do you think prevent you from having a DHCP server? I use VLANs as well and I surely have a DHCP server for each VLAN.

Stuart
